### PR TITLE
removed references of Okteto Cloud

### PR DESCRIPTION
**Pull request by RinkiyaKeDad**
Signed-off-by: Arsh Sharma <arshsharma461@gmail.com>

For the upcoming pricing model changes.

Changes:

- Removed the `/cloud` page and added redirects to `/docs/reference/development-environments/` instead
- Added list of features (from the `/cloud` page) and wrote new description for the `/docs/reference/development-environments/` page
- Replcaed all instances of "Okteto Cloud" with "Okteto"

### CODE REVIEW

File: src/content/cloud/build.mdx [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
 
-The Okteto Build service allows you to build your development images directly in the Okteto Cloud infrastructure. Another reason to remove Docker and Kubernetes from your laptop and develop directly in Okteto Cloud.
+The Okteto Build service allows you to build your development images directly in the Okteto infrastructure. Another reason to remove Docker and Kubernetes from your laptop and develop directly in Okteto.
 
 ## Build images
 
 You'll need the [okteto CLI](/docs/cloud/okteto-cli/) in order to access the Okteto Build Service.
 
-Once the Okteto CLI is installed, run `okteto context` and select the Okteto Cloud option to download the credentials and certificates required to access the Okteto Build service.
+Once the Okteto CLI is installed, run `okteto context` and select the Okteto option to download the credentials and certificates required to access the Okteto Build service.
```
Comment (pchico83) [inline src/content/cloud/build.mdx:16]: When talking about contexts, we should keep Okteto Cloud

File: src/content/cloud/custom-domains.mdx [hunk 2]
```diff
@@ -1,46 +1,46 @@
 ---
 title: Custom Domain Names for your Applications
-description: Use a custom domain name for your applications with Okteto Cloud
+description: Use a custom domain name for your applications with Okteto
```
Comment (pchico83) [inline src/content/cloud/custom-domains.mdx:3]: This document is specific to Okteto Cloud, let's undo changes here

**Review by pchico83 — APPROVED** [review]

**Comment by rberrelleza** [comment]
This is too big to be reviewed effectively. I'd suggest we start just with the current version, and not do the older versions until we decide where is correct to use 'Okteto Cloud' and where we can just mention 'Okteto'.

**Comment by rberrelleza** [comment]
@RinkiyaKeDad could you send a separate PR with the first two items of the PR? Let's hold on the rest until we have decided on the new naming scheme.

**Comment by RinkiyaKeDad** [comment]
Closing this in favour of #107

**Pull request closed without merging**